pod/main/forms.py

```diff
@@ -86,7 +91,8 @@ def __init__(self, request, *args, **kwargs):
         self.fields = add_placeholder_and_asterisk(self.fields)
 
     def clean(self):
+        """Clean "contact us" form submission."""
         cleaned_data = super(ContactUsForm, self).clean()
-        if cleaned_data["subject"] == "-----":
+        if "subject" in cleaned_data and cleaned_data["subject"] == "-----":
```

Comment:
Il vaudrait mieux mettre if not subject in clean data or ("subject" in cleaned_data and cleaned_data["subject"] == "-----"):

Reply:
Concretement, l'erreur "Please specify a subject" ne conviendrait pas dans ce cas, car un subject a bien été spécifié. C'est juste qu'il ne fait pas partie de la liste SUBJECT_CHOICES et a donc été supprimé par Django.

Reply:
Aucun humain ne peut tomber dans ce cas de figure d'ailleurs, puisqu'il faut traffiquer la page html pour remplacer l'input de type "select" par un input type text pour tomber dans ce cas

Reply:
ok, en lisant mieux le code, j'ai vu que je sujet etait obligatoire donc le fait de tester sa valeur n'est même pas utile il me semble puisque elle est vide (et non renseignée avec "-----")

Reply:
mais une verif côté serveur serait quand même utile, on devrait vérifier que la valeur renvoyée est bien dans celle proposée !

Reply:
--> https://pod.univ-lille.fr/contact_us/

Reply:
> mais une verif côté serveur serait quand même utile, on devrait vérifier que la valeur renvoyée est bien dans celle proposée !

c'est deja fait par Django ^^. c'est le super().clean()

Reply:
et quand ce n'est pas dans les champs proposé, il la supprime. d'ou l'erreur "Key error subject not in cleaned_data["subject"]" que je corrige ici ^^

Reply:
en effet, ok pour moi, je corrige ma review